### PR TITLE
Msgpack jruby

### PR DESCRIPTION
**Pull request by tagomoris**
This is follower of #33 

I've done that:
* move specs of msgpack-jruby into specs/jruby
* separate specs about `MessagePack::Packer`, `Unpacker` and `Buffer` into `specs/cruby` as platform-specified specs
* fix some code of java(jruby) implementation
  * de-optimize Float/String size detection to work just same as msgpack-ruby original code
  * add symbolize_keys option of `MessagePack.unpack`

All specs runs with this branch HEAD in success.
And I tested simply in my laptop, jruby runtime and pry. It seems works well.
```
$ gem i Documents/msgpack-ruby/pkg/msgpack-0.5.9-java.gem 
Successfully installed msgpack-0.5.9-java
1 gem installed
$ pry
[1] pry(main)> require 'msgpack'
=> true
[2] pry(main)> MessagePack.dump({"hoge" => 1, "pos" => true})
=> "\x82\xA4hoge\x01\xA3pos\xC3"
[3] pry(main)> MessagePack.unpack(MessagePack.pack({"hoge" => 1, "pos" => true}))
=> {"hoge"=>1, "pos"=>true}
[4] pry(main)> MessagePack.unpack(MessagePack.pack({"hoge" => 1, "pos" => true}), symbolize_keys: true)
=> {:hoge=>1, :pos=>true}
[5] pry(main)> true.to_msgpack
=> "\xC3"
```

We need performance/stress tests, but I think that we can release the version including this code, because many of jruby and msgpack users are using `msgpack-jruby` gem currently.

### CODE REVIEW

**Comment by iconara** [comment]
Great job :+1:

**Comment by tagomoris** [comment]
Tests fails on ruby 1.8.7... Is it still supported?

File: lib/msgpack.rb [hunk 1]
```diff
@@ -1,6 +1,13 @@
 require "msgpack/version"
-begin
-  require "msgpack/#{RUBY_VERSION[/\d+.\d+/]}/msgpack"
-rescue LoadError
-  require "msgpack/msgpack"
+
+if defined?(RUBY_ENGINE) && RUBY_ENGINE == 'jruby'
+  require 'jruby'
```
Comment (frsyuki) [inline lib/msgpack.rb]: Do you mean `require 'java'`? I haven't seen `require 'jruby'`

**Comment by frsyuki** [comment]
Awesome job!
I added a comment on github. "jruby-19mode" test is failing on Travis CI. Could you check it?

Why don't you guys join the msgpack-ruby committer group? It seems hard for me to keep Java code maintained. If both or either of you are happy with it, I think the first work as a committer is to merge this pull-req.

File: lib/msgpack.rb (hunk 1 above)
Comment (tagomoris) [inline lib/msgpack.rb]: `require "jruby"` also works well and works as same with `require "java"`.
But it seems to be `require "java"` for consistency. I'll fix it.

**Comment by tagomoris** [comment]
Tests of "jruby-19mode" fails because that actually runs as JRuby 9000. I confirmed it with jruby-9000-dev on my laptop.
We'll get correct test result tomorrow. https://twitter.com/travisci/status/552465304442634240

**Comment by tagomoris** [comment]
I welcome to join msgpack-ruby committer group to maintain jruby implementation after merged :)

File: lib/msgpack.rb (hunk 1 above)
Comment (iconara) [inline lib/msgpack.rb]: There was a specific reason for `require 'jruby'`: the line below uses `JRuby.runtime`, which is defined in `jruby.rb`. I think neither `require 'jruby'` nor `require 'java'` is needed in newer JRuby versions (1.7+), they're required by default, that's why it appears to work just as well with `require 'java'`.

**Comment by iconara** [comment]
If you'll have me I can join the committer group as well, but as you've seen I haven't had much time to work on this the last six months. My company and me has a strong investment in JRuby and MessagePack though, so it is in our interest to make sure it's the best it can be.

**Comment by tagomoris** [comment]
Very great :+1: for @iconara as committer.

**Comment by tagomoris** [comment]
Green!

**Comment by frsyuki** [comment]
@tagomoris @iconara Awesome! I have sent invitation to both of you. Welcome to msgpack-ruby.

@kosaki @muga @koichiro I've added above 2 people to Ruby Master team.

@iconara I understand that you're busy for other works as well. Because msgpack-ruby is a OSS product, please join discussion or respond/merge pull-requests when you have time.

**Comment by iconara** [comment]
@frsyuki thank you!

**Pull request merged**